src/portable/ehci/ehci.c

```diff
@@ -418,32 +427,36 @@ bool hcd_edpt_xfer(uint8_t rhport, uint8_t dev_addr, uint8_t ep_addr, uint8_t *
 {
   (void) rhport;
 
+  ehci_qhd_t* p_qhd;
+  ehci_qtd_t* p_qtd;
   uint8_t const epnum = tu_edpt_number(ep_addr);
   uint8_t const dir   = tu_edpt_dir(ep_addr);
 
+  // Flush data cache if required, see hcd.h for details
+  hcd_dcache_flush((uint32_t)buffer, buflen);
```

Comment:
we should both invalidate (for rx) and flush (tx) this buffer. In general, I think we should move this dcache flush + invalidate into qtd_init() function.

Reply:
As above...
> I will check this, I am still getting to grips with the USB protocol in general and with the code base, I haven't had time to extensively study the code yet, so please bare with me. 😊